### PR TITLE
Reduce no attestation in pool to warn

### LINKED ISSUE #7859: Validator error level message should be a warning
# 🐞 Bug Report

### Description

```json
{"message":"Could not submit slot signature to beacon node: rpc error: code = Internal desc = Could not find attestation for slot and committee in pool","prefix":"validator","severity":"ERROR","slot":4672}
```

Also, we should return a different error code. I recommend [codes.NotFound](https://github.com/grpc/grpc-go/blob/78864797b87780af8d88b2d4d64721020abbe7ea/codes/codes.go#L73).

### Has this worked before in a previous version?

Probably not. 

## 🔬 Minimal Reproduction

Request to aggregate a signature when there are no attestations in the pool

## 🔥 Error

<pre><code>
{"message":"Could not submit slot signature to beacon node: rpc error: code = Internal desc = Could not find attestation for slot and committee in pool","prefix":"validator","severity":"ERROR","slot":4672}
</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
linux amd64
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
beta.3
  </code>
</pre>

**Anything else relevant (validator index / public key)?**

pyrmont

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Uncomment one line below and remove others.
>
> Clean up

**What does this PR do? Why is it needed?**
When there's no attestation in the pool, aggregator won't be able to submit aggregated attestation. This PR changes status code to unavailable and reduce log from error to warn.

**Which issues(s) does this PR fix?**

Fixes #7859

**Other notes for review**

### CODE REVIEW

File: beacon-chain/rpc/validator/aggregator.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func (vs *Server) SubmitAggregateSelectionProof(ctx context.Context, req *ethpb.
 	if len(aggregatedAtts) == 0 {
 		aggregatedAtts = vs.AttPool.UnaggregatedAttestationsBySlotIndex(req.Slot, req.CommitteeIndex)
 		if len(aggregatedAtts) == 0 {
-			return nil, status.Errorf(codes.Internal, "Could not find attestation for slot and committee in pool")
+			return nil, status.Errorf(codes.Unavailable, "Could not find attestation for slot and committee in pool")
```
Comment (prestonvanloon) [inline beacon-chain/rpc/validator/aggregator.go:71]: This isn't appropriate. Unavailable signals to the client that it should retry later. It's also the default when there is no beacon node serving a request.

What about NotFound as i recommended in the issue description https://github.com/prysmaticlabs/prysm/issues/7859?

Comment (terencechain) [reply beacon-chain/rpc/validator/aggregator.go]: ops, i meant to use `NotFound` here... Fixed. Thanks!

File: validator/client/aggregate.go [hunk 2]
```diff
@@ -66,10 +68,16 @@ func (v *validator) SubmitAggregateAndProof(ctx context.Context, slot uint64, pu
 		SlotSignature:  slotSig,
 	})
 	if err != nil {
-		log.WithField("slot", slot).Errorf("Could not submit slot signature to beacon node: %v", err)
-		if v.emitAccountMetrics {
-			ValidatorAggFailVec.WithLabelValues(fmtKey).Inc()
+		status, ok := status.FromError(err)
+		if ok && status.Code() == codes.NotFound {
+			log.WithField("slot", slot).Warnf("Could not find attestation to submit: %v", err)
```
Comment (prestonvanloon) [inline validator/client/aggregate.go:73]: What do you think about this?
```suggestion
			log.WithField("slot", slot).Warnf("No attestations to aggregate: %v", err)
```

File: validator/client/aggregate.go [hunk 3]
```diff
@@ -66,10 +68,16 @@ func (v *validator) SubmitAggregateAndProof(ctx context.Context, slot uint64, pu
...
+		} else {
+			log.WithField("slot", slot).Errorf("Could not submit slot signature to beacon node: %v", err)
```
Comment (prestonvanloon) [inline validator/client/aggregate.go:75]:
```suggestion
			log.WithField("slot", slot).WithError(err).Error("Could not submit slot signature to beacon node")
```

File: validator/client/aggregate.go (hunk 2 above)
Comment (terencechain) [reply validator/client/aggregate.go]: love it, thanks @prestonvanloon !

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**